mbuild/formats/lammpsdata.py

```diff
@@ -136,14 +139,48 @@ def write_lammpsdata(structure, filename, atom_style='full'):
             data.write('{:d}\t{:.6f}\t# {}\n'.format(atom_type,mass,unique_types[atom_type-1]))
 
         if forcefield:
-            # Pair coefficients
-            epsilons = [atom.epsilon for atom in structure.atoms]
             sigmas = [atom.sigma for atom in structure.atoms]
-            epsilon_dict = dict([(unique_types.index(atom_type)+1,epsilon) for atom_type,epsilon in zip(types,epsilons)])
+            epsilons = [atom.epsilon for atom in structure.atoms]
             sigma_dict = dict([(unique_types.index(atom_type)+1,sigma) for atom_type,sigma in zip(types,sigmas)])
-            data.write('\nPair Coeffs # lj\n\n')
-            for idx,epsilon in epsilon_dict.items():
-                data.write('{}\t{:.5f}\t{:.5f}\n'.format(idx,epsilon,sigma_dict[idx]))
+            epsilon_dict = dict([(unique_types.index(atom_type)+1,epsilon) for atom_type,epsilon in zip(types,epsilons)])
+
+            # Modified cross-interactions
+            if structure.has_NBFIX():
+                params = ParameterSet.from_structure(structure)
+                warn('Explicitly writing cross interactions using mixing rule: {}'.format(
+                    structure.combining_rule))
+                coeffs = dict()
+                for combo in it.combinations_with_replacement(unique_types, 2):
+                    # Attempt to find pair coeffis in nbfixes
+                    if combo in params.nbfix_types:
+                        type1 = unique_types.index(combo[0])+1
+                        type2 = unique_types.index(combo[1])+1
+                        rmin = params.nbfix_types[combo][0] # Angstrom
+                        epsilon = params.nbfix_types[combo][1] # kcal
+                        sigma = rmin/2**(1/6)
+                        coeffs[(type1, type2)] = (sigma, epsilon)
+                    else:
+                        type1 = unique_types.index(combo[0]) + 1
+                        type2 = unique_types.index(combo[1]) + 1
+                        # Might not be necessary to be this explicit
+                        if type1 == type2:
+                            sigma = sigma_dict[type1]
+                            epsilon = epsilon_dict[type1]
+                        else:
+                            sigma = (sigma_dict[type1]+sigma_dict[type2])*0.5
```

Comment:
Need to check to see if `Structure.combining_rule` is `lorentz` or `geometric` which will determine whether you need to take an arithmetic or geometric average here.